### PR TITLE
kotlin-native: update darwin hashes; fix build

### PR DESCRIPTION
**Pull request by ghost**
the last version auto update https://github.com/NixOS/nixpkgs/pull/264833 only updated the linux x64 hash and neglected the 2 macos ones. the auto update will likely break again soon as the next version was [released](https://github.com/JetBrains/kotlin/releases/tag/v1.9.21) 5 days ago

this updates the other two hashes using the [published](https://github.com/JetBrains/kotlin/releases/tag/v1.9.20)  sha256 digests and converts them to an sri.
```
# aarch64 darwin
nix-hash --to-sri --type sha256 630e883bca63d02f571dfcd5b21728be4a050545c9dca8b1846a326c14452f65
# x64 darwin
nix-hash --to-sri --type sha256 9a70e2f1d3020f31ac74f743d5dc3860c07101fa3916fc0325553f7b0ef4cfd5
```

testing: build passed on x64 darwin

ZHF: #265948
https://hydra.nixos.org/build/240783888

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by fabianhjr** [comment]
Hi, thanks for the fix, I apologize for the issues caused since I should have looked into the diff before merging.

I haven't been using kotlin lately, would you be interested in adding yourself as a maintainer of this package? (You would be notified on automatic updates)

**Comment by a-n-n-a-l-e-e** [comment]
> Hi, thanks for the fix, I apologize for the issues caused since I should have looked into the diff before merging.
> 
> I haven't been using kotlin lately, would you be interested in adding yourself as a maintainer of this package? (You would be notified on automatic updates)

no worries. i don't use kotlin either. i've just been fixing broken packages for the upcoming release. #265948

**Comment by fabianhjr** [comment]
Thanks for the help with ZHF, merging as is then. :3

**Review by fabianhjr — APPROVED** [review]

**Pull request merged**
